gc/gc-base/src/main/java/org/projectnessie/gc/base/GCImpl.java

```diff
@@ -115,14 +120,15 @@ public String identifyExpiredContents(SparkSession session) {
           gcParams.getBloomFilterExpectedEntries() == null
               ? getTotalCommitsInDefaultReference(api)
               : gcParams.getBloomFilterExpectedEntries();
-      // Identify the live contents and return the bloom filter per content-id
-      liveContentsBloomFilterMap =
-          distributedIdentifyContents.getLiveContentsBloomFilters(
-              allRefs, bloomFilterSize, droppedReferenceTimeMap);
-      getOrCreateEmptyBranch(api, gcParams.getOutputBranchName());
+      GCUtil.getOrCreateEmptyBranch(api, gcParams.getOutputBranchName());
+      // Identify the live contents and return the bloom filter
+      liveContentsBloomFilter =
+          distributedIdentifyContents.getLiveContentsBloomFilter(
+              allRefs, bloomFilterSize, droppedReferenceTimeMap, runId, startedAt);
     }
     // Identify the expired contents
-    return distributedIdentifyContents.identifyExpiredContents(liveContentsBloomFilterMap, allRefs);
+    return distributedIdentifyContents.identifyExpiredContents(
```

Comment:
This call will use the **closed** `NessieApiV1`!

Reply:
sorry, I don't see where I am using `NessieApiV1` in driver after line:128. 

The executor will build its own `NessieApiV1`.  So, I still don't see the problem here.